### ISSUE TITLE
Broken Tests on macOS 10.15

### ISSUE THREAD

**Issue by samdoran** [ansible/ansible#71836]
##### SUMMARY
A macOS 10.15 image is now available for testing in CI. Since we have been testing on OS X 10.11 up until now, there are a number of broken tests that need to be fixed before we can switch to using the macOS 10.15 image.

##### ISSUE TYPE
- Bug Report

##### COMPONENT NAME

`test/integration/targets/become`
`test/integration/targets/get_url`
`test/integration/targets/hostname`
`test/integration/targets/interpreter_discovery_python`
`test/integration/targets/lookup_url`
`test/integration/targets/service`
`test/integration/targets/user`
`test/integration/targets/facts_linux_network`
`test/integration/targets/filter_core`
`test/integration/targets/subversion`
`test/integration/targets/unarchive`
`test/integration/targets/binary_modules_posix`
`test/integration/targets/connection_paramiko_ssh`
`test/integration/targets/package_facts`
`test/integration/targets/remote_tmp`
`test/integration/targets/service_facts`
`test/integration/targets/ansible-galaxy`
`test/integration/targets/ansible-galaxy-collection-scm`
`test/integration/targets/cron`
`test/integration/targets/git`
`test/integration/targets/uri`
`test/integration/targets/pip`

##### ANSIBLE VERSION

```paste below
devel
```

##### CONFIGURATION

```paste below
Default
```

##### OS / ENVIRONMENT

Managed node:
```
ProductName:	Mac OS X
ProductVersion:	10.15.6
BuildVersion:	19G2021
```

```
Python 3.8.5
```

##### STEPS TO REPRODUCE
Select a broken test from the checklist below. Run the test with the following command and fix the failure. Then submit a PR with the fix against `devel`. Once that fix is merged, backport it to `stable-2.10`

```
ansible-test integration --remote macos/10.15 [target] -v
```

##### EXPECTED RESULTS

Tests pass

##### ACTUAL RESULTS
Tests fail. See https://app.shippable.com/github/ansible/ansible/runs/173960/ for example failures.

###### Group 1

- [ ] become
- [ ] get_url
- [ ] hostname
- [ ] interpreter_discovery_python
- [ ] lookup_url
- [ ] service
- [ ] user

###### Group 2

- [ ] facts_linux_network
- [ ] filter_core
- [ ] subversion
- [ ] unarchive

###### Group 3

- [ ] binary_modules_posix
- [ ] connection_paramiko_ssh
- [ ] package_facts
- [ ] remote_tmp
- [ ] service_facts

###### Group 4

- [ ] ansible-galaxy
- [ ] ansible-galaxy-collection-scm
- [ ] cron
- [ ] git
- [ ] uri

###### Group 5

- [ ] pip

**Comment by samdoran** [comment]
Once  all these tests are fixed and we switch to using the new image, https://github.com/ansible/ansible/pull/67289 can be reverted.

**Comment by sivel** [comment]
##### filter_core
Not working due to missing `passlib`, this is in all containers

##### get_url
`Destination /usr/local/etc/openssl/certs does not exist`

I cannot run this myself for some reason, the http-test-container isn't accessible.

Paths used by the Python installation:

```json
{
    "paths": [
        "/private/etc/ssl/cert.pem",
        "/private/etc/ssl/certs",
        "/etc/ansible",
        "/usr/local/etc/openssl"
    ]
}
```

**Comment by mattclay** [comment]
@samdoran I don't think we'll want to revert https://github.com/ansible/ansible/pull/67289 at any point, since it's necessary for use of `osx/10.11` -- we could consider removing the same constraint from the `macos` portion of the setup script though.

**Comment by sivel** [comment]
We'll need to add `skip/macos` to all of these targets:

```
test/integration/targets/ansible-runner/aliases:4:skip/osx
test/integration/targets/apt/aliases:4:skip/osx
test/integration/targets/apt_key/aliases:3:skip/osx
test/integration/targets/apt_repository/aliases:4:skip/osx
test/integration/targets/connection_delegation/aliases:3:skip/osx  # No sshpass
test/integration/targets/cron/aliases:4:skip/osx
test/integration/targets/dnf/aliases:6:skip/osx
test/integration/targets/dpkg_selections/aliases:5:skip/osx
test/integration/targets/facts_linux_network/aliases:4:skip/osx
test/integration/targets/hostname/aliases:4:skip/osx  # same, see #54439, #32214
test/integration/targets/incidental_cloud_init_data_facts/aliases:4:skip/osx
test/integration/targets/incidental_flatpak_remote/aliases:5:skip/osx
test/integration/targets/incidental_inventory_docker_swarm/aliases:4:skip/osx
test/integration/targets/incidental_mongodb_parameter/aliases:4:skip/osx
test/integration/targets/incidental_setup_zabbix/aliases:3:skip/osx
test/integration/targets/incidental_timezone/aliases:4:skip/osx
test/integration/targets/incidental_ufw/aliases:4:skip/osx
test/integration/targets/incidental_zabbix_host/aliases:4:skip/osx
test/integration/targets/module_no_log/aliases:4:skip/osx  # not configured to log user.info to /var/log/syslog
test/integration/targets/old_style_cache_plugins/aliases:2:skip/osx
test/integration/targets/package_facts/aliases:3:skip/osx
test/integration/targets/service/aliases:4:skip/osx
test/integration/targets/service_facts/aliases:4:skip/osx
test/integration/targets/subversion/aliases:4:skip/osx
test/integration/targets/yum/aliases:5:skip/osx
```

**Comment by samdoran** [comment]
Fixed by #71841
